neuraxle/base.py

```diff
@@ -29,14 +29,55 @@
 from neuraxle.hyperparams.space import HyperparameterSpace, HyperparameterSamples
 
 
-class BaseStep(ABC):
+class Hasher(ABC):
```

Comment:
perhaps use a mixin/wrapper to wrap the hasher with the hyperparams hashing method ?